diff --color -Naur a/meson_options.txt b/meson_options.txt
--- a/meson_options.txt	2024-08-29 00:13:18.000000000 +0200
+++ b/meson_options.txt	2025-12-23 13:16:03.649931442 +0100
@@ -140,6 +140,13 @@
 )
 
 option(
+  'gallium-vc4-neon',
+  type : 'feature',
+  deprecated: {'true': 'enabled', 'false': 'disabled'},
+  description : 'enable gallium vc4 optional neon support.',
+)
+
+option(
   'va-libs-path',
   type : 'string',
   value : '',
diff --color -Naur a/src/gallium/drivers/vc4/meson.build b/src/gallium/drivers/vc4/meson.build
--- a/src/gallium/drivers/vc4/meson.build	2024-08-29 00:13:18.000000000 +0200
+++ b/src/gallium/drivers/vc4/meson.build	2025-12-23 13:17:54.366459188 +0100
@@ -89,7 +89,7 @@
 ]
 
 libvc4_neon = []
-if host_machine.cpu_family() == 'arm'
+if host_machine.cpu_family() == 'arm' and not get_option('gallium-vc4-neon').disabled()
   libvc4_neon = static_library(
     'vc4_neon',
     'vc4_tiling_lt_neon.c',
@@ -99,7 +99,7 @@
     c_args : '-mfpu=neon',
     dependencies : vc4_deps,
   )
-  vc4_c_args += '-DUSE_ARM_ASM'
+  vc4_c_args += '-DVC4_TILING_LT_NEON'
 endif
 
 if dep_simpenrose.found()
diff --color -Naur a/src/gallium/drivers/vc4/vc4_tiling.h b/src/gallium/drivers/vc4/vc4_tiling.h
--- a/src/gallium/drivers/vc4/vc4_tiling.h	2024-08-29 00:13:18.000000000 +0200
+++ b/src/gallium/drivers/vc4/vc4_tiling.h	2025-12-23 13:20:02.489912672 +0100
@@ -89,7 +89,7 @@
                   void *src, uint32_t src_stride,
                   int cpp, const struct pipe_box *box)
 {
-#ifdef USE_ARM_ASM
+#ifdef VC4_TILING_LT_NEON
         if (util_get_cpu_caps()->has_neon) {
                 vc4_load_lt_image_neon(dst, dst_stride, src, src_stride,
                                        cpp, box);
@@ -105,7 +105,7 @@
                    void *src, uint32_t src_stride,
                    int cpp, const struct pipe_box *box)
 {
-#ifdef USE_ARM_ASM
+#ifdef VC4_TILING_LT_NEON
         if (util_get_cpu_caps()->has_neon) {
                 vc4_store_lt_image_neon(dst, dst_stride, src, src_stride,
                                         cpp, box);
